gr-blocks/lib/tagged_file_sink_impl.cc

```diff
@@ -63,12 +63,8 @@ tagged_file_sink_impl::tagged_file_sink_impl(size_t itemsize, double samp_rate)
                  io_signature::make(1, 1, itemsize),
                  io_signature::make(0, 0, 0)),
       d_itemsize(itemsize),
-      d_n(0),
       d_sample_rate(samp_rate)
 {
-    d_state = NOT_IN_BURST;
```

Comment:
I can see why these would get tagged as an issue ... why not move these into the constructor instantiations above instead?

Reply:
If you prefer. I suppose this belongs in https://github.com/ThomasHabets/greps/blob/grep0018/grep-0018-modernize-c%2B%2B.md

Reply:
Hmmm ... not sure these are "modern" C++ ... thinking this is just reasonable C++ programming, not specific to C++11 or newer even.

Reply:
Declaration initialization is C++11.

Reply:
My preference is for default values at declare time (that is, in header files). Two other alternatives (in the presence of multiple constructors at least) is duplicate code (our current state), or let constructors call constructors (also new in C++11).

Reply:
> Declaration initialization is C++11.

Ah OK. that explains why I don't use them yet! I'm -so- behind on my C++ syntax!

Reply:
OK so what if there's a Declaration initialization as well as a constructor initialization of the same variable ... is that "bad" coding style? Which setting "wins"? Guessing the constructor initialization ... if the compiler allows it ... hmmm ....

Reply:
OK , so yes, I'd say to add this to your GREP ... now that I know/understand what it's all about :)

Reply:
> OK so what if there's a Declaration initialization as well as a constructor initialization of the same variable ... is that "bad" coding style? Which setting "wins"? Guessing the constructor initialization ... if the compiler allows it ... hmmm ....

The compiler allows it and the constructor wins, see [1]. There, the term _default member initializer_ is used instead of _declaration initialization_. You better not look up what other kinds of initializations exist in C++ ;)

[1] https://en.cppreference.com/w/cpp/language/data_members#Member_initialization

Reply:
oh lordy!